lib/adapters/outline-view-adapter.js

```diff
@@ -27,9 +31,25 @@ export default class OutlineViewAdapter {
   //
   // Returns a {Promise} containing the {Outline} of this document.
   async getOutline(connection: LanguageClientConnection, editor: atom$TextEditor): Promise<?atomIde$Outline> {
+    let cancellationToken = this._cancellationTokens.get(connection);
+    if (cancellationToken !== undefined) {
+      cancellationToken.cancel();
+    }
+
+    cancellationToken = new CancellationTokenSource();
+    this._cancellationTokens.set(connection, cancellationToken);
+
     const results = await connection.documentSymbol({
       textDocument: Convert.editorToTextDocumentIdentifier(editor),
-    });
+    }, cancellationToken.token);
+
+    this._cancellationTokens.delete(connection);
+    if (cancellationToken.token.isCancellationRequested) {
+      return {
+        outlineTrees: [],
+      };
+    }
+
```

Comment:
Wonder if we can combine this logic with the same logic in the autocomplete adapter into a reusable function/chunk?

Reply:
I'll give it a shot!